packages/libgrib1/package.py

```diff
@@ -35,40 +35,29 @@ class Libgrib1(MakefilePackage):
 
     version('master', branch='master')
     version('22-01-2020', commit='3d3db9a9a090f6798c2fd4290c271dd58ff694e0')
-    
-    variant('slave', default='tsa', description='Build on slave tsa, daint or kesch', multi=False)
-
-    depends_on('mpi')
-
-    def setup_environment(self, spack_env, run_env):
-        spack_env.set('LIBNAME', 'grib1')
+   
+    def edit(self, spec, prefix):
+        _makefile_name = 'Makefile.linux'
+        if self.compiler.name == 'gcc':
+          _makefile_name += '.gnu'
+        elif self.compiler.name == 'pgi':
+            _makefile_name += '.pgi'
+        elif self.compiler.name == 'cce':
+            _makefile_name += '.cray'
+        self._makefile_name = _makefile_name
 
     def build(self, spec, prefix):
         with working_dir(self.build_directory):
-            MakeFileName = 'Makefile.' + self.spec.variants['slave'].value
-            if self.compiler.name == 'gcc':
-                MakeFileName += '.gnu'
-            elif self.compiler.name == 'pgi':
-                MakeFileName += '.pgi'
-            elif self.compiler.name == 'cce':
-                MakeFileName += '.cray'
-            MakeFileFilter = FileFilter(MakeFileName)
+            MakeFileFilter = FileFilter(self._makefile_name)
```

Comment:
local variables are usually snake_case as well, but since it's only a local, it doesn't really matter and we're already losing too much time due to CamelCase vs snake_case.